ddtrace/utils/__init__.py

```diff
@@ -25,3 +25,13 @@ def __get__(self, cls, owner):
             "Usage of ddtrace.ext.AppTypes is not longer supported, please use ddtrace.ext.SpanTypes"
         )
         return classmethod(self.fget).__get__(None, owner)()
+
+
+def external_service(config, pin):
+    if pin.service:
+        return pin.service
+    elif config.service:
+        return config.service
+    elif config.service_name:
```

Comment:
why do we have both `service` and `service_name`?

Reply:
seems like this could just be:

```
def external_service(config, pin):
    return pin.service or config.service
```

?

Reply:
We do unfortunately 😕